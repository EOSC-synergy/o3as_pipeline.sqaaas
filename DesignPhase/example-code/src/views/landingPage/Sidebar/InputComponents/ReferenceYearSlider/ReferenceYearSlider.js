import React, { useState } from "react";

/**
 * enables the user to select a reference year
 * @param {*} props 
 *  props.defaultReferenceYear -> a int containing the default reference year
 *  props.min -> minimum value for reference year
 *  props.max -> maximum value for reference year
 * @returns a jsx containing a slider to select the reference year
 */
function ReferenceYearSlider(props) {

    let i = props.min;
    i = props.max;
    i = props.error;

    // TODO -> redux
    i = props.currReferenceYear;
    i = props.changeReferenceYear;

    return (
        <>
        </>
    );
}

export default ReferenceYearSlider;