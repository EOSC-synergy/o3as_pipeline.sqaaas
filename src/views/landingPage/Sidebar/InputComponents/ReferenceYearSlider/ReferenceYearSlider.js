import React, { useState } from "react";
import { Grid, Typography, Slider, MenuItem } from "@mui/material";
import models from "./models.json";
import { Select, InputLabel, OutlinedInput, FormControl } from "@mui/material";

/**
 * enables the user to select a reference year
 * @todo add redux connection: should min and max also be in redux?
 * @param {Object} props
 * @param {function} props.reportError - function to handle errors
 * @returns {JSX} a jsx containing a slider to select the reference year
 */
function ReferenceYearSlider(props) {
  let i = props.reportError;

  /** model: The currently selected reference model.
   * setModel: The function which adjusts the SelectionBar after selecting a model.
   */
  const [model, setModel] = React.useState([]);

  /** Default year value for the Slider.*/
  const defaultYear = 1980;

  /** Minimum chosable year value for the Slider.*/
  const minYear = 1950;

  /** Minimum chosable year value for the Slider.*/
  const maxYear = 2100;

  /** Handles the change of the reference model selection if it's is modified.*/
  const handleChangeForRefModel = (event) => {
    const {
      target: { value },
    } = event;
    setModel(typeof value === "string" ? value.split(",") : value);
    // TODO
    //dispatch(setModel({title: event.target.value}));
  };

  /** Handles the change of the reference year slider if it's is modified.*/
  const handleChangeForRefYear = (event) => {
    const {
      target: { value },
    } = event;

    // TODO
    //dispatch(setYear({title: event.target.value}));
  };

  return (
    <>
      <FormControl sx={{ m: 1, width: "100%" }}>
        <Select
          labelId="locationSelectLabel"
          id="locationSelect"
          label="Reference Model"
          onChange={handleChangeForRefModel}
          input={<OutlinedInput label="Reference Model" />}
          defaultValue={1}
          value={model}
        >
          {models.map((elem, idx) => {
            return (
              <MenuItem key={idx} value={elem}>
                {elem}
              </MenuItem>
            );
          })}
        </Select>
        <InputLabel id="locationSelectLabel">Reference Model</InputLabel>
      </FormControl>

      <Grid container>
        <Grid item xs={4}>
          <Typography>Reference year:</Typography>
        </Grid>
        <Grid item xs={8}>
          <Slider
            defaultValue={defaultYear}
            step={1}
            min={minYear}
            max={maxYear}
            onChange={handleChangeForRefYear}
            valueLabelDisplay="on"
            size="small"
            track={false}
          />
        </Grid>
      </Grid>
    </>
  );
}

export default ReferenceYearSlider;
