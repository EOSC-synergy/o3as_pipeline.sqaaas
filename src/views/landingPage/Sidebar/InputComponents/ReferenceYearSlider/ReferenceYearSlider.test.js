import React from 'react';
import ReactDOM from 'react-dom';
import ReferenceYearSlider from './ReferenceYearSlider';
import { render } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

it('Component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ReferenceYearSlider />, div);
});

// Snapshot test
it('Component renders correctly from config file', () => {
    const { container } = render(<ReferenceYearSlider />);
    expect(container).toMatchSnapshot();
});
