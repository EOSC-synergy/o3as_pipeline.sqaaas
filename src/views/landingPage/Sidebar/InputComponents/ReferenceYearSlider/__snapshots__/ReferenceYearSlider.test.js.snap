// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly from config file 1`] = `
<div>
  <div
    class="MuiFormControl-root css-78me8v-MuiFormControl-root"
  >
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="locationSelectLabel locationSelect"
        class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="locationSelect"
        role="button"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          â€‹
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-173wfxe"
        >
          <span>
            Reference Model
          </span>
        </legend>
      </fieldset>
    </div>
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      id="locationSelectLabel"
    >
      Reference Model
    </label>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Reference year:
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
    >
      <span
        class="MuiSlider-colorPrimary MuiSlider-sizeSmall MuiSlider-root MuiSlider-trackFalse css-53tbrc-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1vjujtc-MuiSlider-track"
          style="left: 0%; width: 20%;"
        />
        <span
          class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeSmall MuiSlider-thumb css-14gf62f-MuiSlider-thumb"
          data-index="0"
          style="left: 20%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="2100"
            aria-valuemin="1950"
            aria-valuenow="1980"
            data-index="0"
            max="2100"
            min="1950"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="1980"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-gdpt3b-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                1980
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;
